regression/run/references/references.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Shroud Project Developers.
+// See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//
+// references.hpp - wrapped routines
+//
+
+#ifndef REFERENCES_HPP
+#define REFERENCES_HPP
+
+class ArrayWrapper
+{
+public:
+
+    /// Default constructor
+    ArrayWrapper() = default;
+
+    /// Simple destructor
+    ~ArrayWrapper() { delete[] arr; }
+
+    /// Sets the size of the array
+    void setSize(int size) { sz = size; }
+
+    /// Sets the size of the array
+    int getSize() const { return sz; }
+
+    /// Allocates the array
+    void allocate() 
+    {
+        delete[] arr;
+        arr = nullptr;
+
+        if(sz >=0 ) 
+            { 
+                arr = new double[sz]; 
+            }
+    }
+
+    /// Access the array
+    double* getArray() { return arr; }
+
```

Comment:
For completeness, it might make sense to also add an accessor for `arr` of the form:
```cpp
void getArrayPtr( double* arr, int sz);   // or would it need to be `double**` ?
```

Reply:
It would also be helpful to have `const` accessors, to show users how they work and to ensure that `shroud` can handle them.
E.g.
```cpp
double* getArray() const { return arr; } 
```

Reply:
It would be `double **arr +intent(out)`.  I think there is a test like that in pointers.yaml but it may be helpful to have them in the same file to compare and contrast.
The pointers.yaml file is used as both C and C++.  I named the new test references.yaml since I intend to add stuff like `double &* +intent(out)` which is not legal in C.